infra/RepoManager.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class to manage a git repository via python.
+
+This class is to be used to implement git commands over
+a python API and manage the current state of the git repo.
+
+  Typical usage example:
+
+    r_man =  RepoManager('https://github.com/google/oss-fuzz.git')
+    r_man.checkout('5668cc422c2c92d38a370545d3591039fb5bb8d4')
+    r_man.getBranch('bisector')
+    r_man.close()
+"""
+import os
+import shutil
+import subprocess
+
+
+class RepoManagerException(Exception):
+  """Class to describe the exceptions in RepoManager."""
+
+  def __init__(self, message):
+    """ Init the exception class.
+
+    Args:
+      message: the message to propigate to user
+    """
+    super().__init__(message)
+
+
+class RepoManager(object):
+  """Class to manage git repos from python.
+
+  Attributes:
+    repo_url: The location of the git repo
+    local_dir: The location of where the repo clone is stored locally
```

Comment:
maybe call it `work_dir`? Otherwise it's not straightforward that this is the directory where we'll create a directory where we clone the repository into

Reply:
or `base_dir`